File: test-suite/src/test/java/io/micronaut/email/docs/SendAttachmentService.java
package io.micronaut.email.docs;

import io.micronaut.email.Attachment;
import io.micronaut.email.BodyType;
import io.micronaut.email.Email;
import io.micronaut.email.EmailSender;
import io.micronaut.email.MultipartBody;
import io.micronaut.email.test.SpreadsheetUtils;
import io.micronaut.http.MediaType;
import jakarta.inject.Singleton;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import java.io.ByteArrayOutputStream;
import java.io.IOException;

@Singleton
public class SendAttachmentService {
    private final EmailSender<?, ?> emailSender;

    public SendAttachmentService(EmailSender<?, ?> emailSender) {
        this.emailSender = emailSender;
    }

    public void sendReport() throws IOException {
        emailSender.send(Email.builder()
                .from("sender@example.com")
                .to("john@example.com")
                .subject("Monthly reports")
                .body(new MultipartBody("<html><body><strong>Attached Monthly reports</strong>.</body></html>", "Attached Monthly reports"))
                .attachment(Attachment.builder()
                        .filename("reports.xlsx")
                        .contentType("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
                        .content(excel())
                        .build()));
    }

    private static byte[] excel() throws IOException {
        XSSFWorkbook wb = new XSSFWorkbook();
        wb.createSheet("Reports");
        ByteArrayOutputStream bos = new ByteArrayOutputStream();
        try {
            wb.write(bos);
        } finally {
            bos.close();
        }
        return bos.toByteArray();
    }
}

File: test-suite-python/src/test/python/micronaut/email/docs/SendAttachmentService.py
from jakarta.inject import Singleton
from java.io import ByteArrayOutputStream
from micronaut.email import Attachment, Email, EmailSender, MultipartBody
from org.apache.poi.xssf.usermodel import XSSFWorkbook


@Singleton
class SendAttachmentService:
    def __init__(self, email_sender: EmailSender):
        self.email_sender = email_sender

    def send_report(self) -> None:
        self.email_sender.send(Email.builder()
                .from_("sender@example.com")
                .to("john@example.com")
                .subject("Monthly reports")
                .body(MultipartBody("<html><body><strong>Attached Monthly reports</strong>.</body></html>", "Attached Monthly reports"))
                .attachment(Attachment.builder()
                        .filename("reports.xlsx")
                        .contentType("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
                        .content(self.excel())
                        .build()))

    @staticmethod
    def excel() -> bytes:
        wb = XSSFWorkbook()
        wb.createSheet("Reports")
        bos = ByteArrayOutputStream()
        try:
            wb.write(bos)
        finally:
            bos.close()
        return bos.toByteArray()
